e2e/utils/waits-utils.ts

```diff
@@ -39,7 +39,7 @@ export async function waitForDocumentTitle(page: Page, titleSubstr: string): Pro
     const jsHandle = await page.waitForFunction(t => {
       const actualTitle = document.title;
       return actualTitle.includes(t);
-    }, {}, titleSubstr);
+    }, {timeout: 30000}, titleSubstr);
```

Comment:
The default timeout for `waitForFunction` is 10 seconds. It is too short and made tests flaky. Increase it to 30 seconds helps to reduce flaky.

Reply:
If you think individual steps are too slow, please file tickets on those, especially if it's a regression.